StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/Vertical Saved Payment Method Screen/PaymentMethodRowButton.swift

```diff
@@ -0,0 +1,131 @@
+    // TODO(porter) Maybe expand this into an enum of (selected, unselected, editing) state
+    var isSelected: Bool {
+        get {
+            return shadowRoundedRect.isSelected
+        }
+
+        set {
+            shadowRoundedRect.isSelected = newValue
+            circleView.alpha = newValue ? 1.0 : 0.0
+        }
+    }
+
+    weak var delegate: PaymentMethodRowButtonDelegate?
+
+    // MARK: Private properties
+    private let viewModel: ViewModel
+    private let height = 44.0 // Hardcoded height from figma
+
+    // MARK: Private views
+
+    private lazy var paymentMethodImageView: UIImageView = {
+        let imageView = UIImageView(image: viewModel.image)
+        imageView.contentMode = .scaleAspectFit
+        // TODO(porter) Do we want to round the corners?
+        return imageView
+    }()
+
+    private lazy var label: UILabel = {
+        let label = UILabel()
+        label.text = viewModel.text
+        label.font = viewModel.appearance.scaledFont(for: viewModel.appearance.font.base.medium,
+                                                     style: .callout,
+                                                     maximumPointSize: 25)
+        label.adjustsFontForContentSizeCategory = true
+        return label
+    }()
+
+    private lazy var circleView: CheckmarkCircleView = {
+        let circleView = CheckmarkCircleView(fillColor: viewModel.appearance.colors.primary)
+        circleView.alpha = 0.0
+        return circleView
+    }()
+
+    private lazy var stackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [paymentMethodImageView, label, UIView.spacerView, circleView])
+        stackView.axis = .horizontal
+        stackView.alignment = .center
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.directionalLayoutMargins = PaymentSheetUI.defaultMargins
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.setCustomSpacing(12, after: paymentMethodImageView) // Hardcoded from figma
+        return stackView
+    }()
+
+    private lazy var shadowRoundedRect: ShadowedRoundedRectangle = {
+        let shadowRoundedRect = ShadowedRoundedRectangle(appearance: viewModel.appearance)
+        shadowRoundedRect.translatesAutoresizingMaskIntoConstraints = false
+        shadowRoundedRect.addAndPinSubview(stackView)
+        return shadowRoundedRect
+    }()
+
+    init(viewModel: ViewModel) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+
+        addSubview(shadowRoundedRect)
+        NSLayoutConstraint.activate([
+            shadowRoundedRect.topAnchor.constraint(equalTo: topAnchor),
+            shadowRoundedRect.leadingAnchor.constraint(equalTo: leadingAnchor),
+            shadowRoundedRect.trailingAnchor.constraint(equalTo: trailingAnchor),
+            shadowRoundedRect.bottomAnchor.constraint(equalTo: bottomAnchor),
+            shadowRoundedRect.heightAnchor.constraint(equalToConstant: height),
```

Comment:
Why is this needed?

Reply:
Without explicitly setting the height of the `shadowRect` it is just sized the minimum to fit the content. Is there a better way to do this?

Removing the `shadowRoundedRect.heightAnchor.constraint(equalToConstant: height)`:
![CleanShot 2024-05-09 at 18 11 17](https://github.com/stripe/stripe-ios/assets/88012362/91b5fec1-add6-4cc4-aa13-49779f69733b)

Reply:
Looks like the stack view doesn't have top or bottom margins.

```
stackView.directionalLayoutMargins = PaymentSheetUI.defaultMargins
```

Reply:
Oh good catch, ty!